service-base/src/test/java/org/eclipse/hono/service/tenant/AbstractCompleteTenantServiceTest.java

```diff
@@ -51,12 +83,13 @@
     public void testAddTenantFailsForDuplicateTenantId(final TestContext ctx) {
 
         addTenant("tenant").map(ok -> {
-            getCompleteTenantService().add(
+            processAddRequest(
```

Comment:
so that all tenant API request will go through the single entry point that all tenant API requests should go through. this entry point is in the abstract base implementation and is responsible for doing the all request validation checks before delegating the request to the concrete implementations

Reply:
sure! but some of the tests uses a simple string value to denote a trusted CA's public key or cert and still succeeds in adding the tenant.

yes i agree, under the assumption that we already know the invalid trusted CA will cause the request to fail. in this case, the creation of a self signed certificate is not really required.

will revert